jax/test_util.py

```diff
@@ -868,3 +871,79 @@ def ignore_warning(**kw):
   with warnings.catch_warnings():
     warnings.filterwarnings("ignore", **kw)
     yield
+
+
+class _cached_property:
```

Comment:
Can you use https://docs.python.org/3/library/functools.html#functools.lru_cache instead of creating your own caching shim?

Reply:
No, unfortunately lru_cache doesn't work on class methods, and I'm not aware of any standard library function that does a similar thing.